File: Product Application/Models/ProductModel.cs
// Define the namespace for the models used in the Product_Application project
namespace Product_Application.Models
{
    // Represents a model for a product, containing various properties related to a product's details
    public class ProductModel
    {
        // Property to store the product ID (nullable to allow for cases where the ID is not yet assigned)
        public int? Id { get; set; }

        // Property to store the status of the product (e.g., "In Stock", "Out of Stock")
        public string Status { get; set; }

        // Property to store the product code, which is typically a unique identifier or SKU
        public string ProductCode { get; set; }

        // Property to store the name of the product
        public string ProductName { get; set; }

        // Property to store the name of the customer associated with the product (if applicable)
        public string CustomerName { get; set; }

        // Property to store the quantity of the product
        public int Quantity { get; set; }

        // Property to store the delivery date of the product
        public DateTime DeliveryDate { get; set; }

        // Default constructor
        public ProductModel()
        {
            // Initialize the properties with default values if needed
        }

        // Constructor that allows setting all properties during instantiation
        public ProductModel(int id, string status, string productCode, string productName, string customerName, int quantity, DateTime deliveryDate)
        {
            Id = id;
            Status = status;
            ProductCode = productCode;
            ProductName = productName;
            CustomerName = customerName;
            Quantity = quantity;
            DeliveryDate = deliveryDate;
        }
    }
}

File: ProductAppAngular/src/app/models/product.model.ts
// Interface to represent the structure of a Product object
export interface Product {

  id?: number; // Optional: Unique identifier for the product
  status: string; // The status of the product (e.g., "In Stock", "Out of Stock")
  productCode: string; // A unique code representing the product
  productName: string; // The name of the product
  customerName: string; // The name of the customer associated with the product
  quantity: number; // The quantity of the product available or ordered
  deliveryDate: Date; // The date when the product is expected to be delivered
}

// Interface to represent the structure of a product request, typically used for filtering or searching products
export interface ProductRequestModel {

  searchString?: string; // Optional: A search string used to filter products by name, code, customer, etc.
}
